### ISSUE TITLE
DAV Mount storage limit

### ISSUE THREAD

**Issue by SilinPavel** [epam/cloud-pipeline#2233]
- It would be great to have opportunity for user to request to mount data storage for dav service automatically (now it only possible if user has permissions to add metadata to storage)
- It also great to have possibility to define time period for the datastorage to be mounted to dav, after this time storage should be automatically unmounted from dav

**Comment by SilinPavel** [comment]
The following approach is proposed for this issue:

### New datastorage API

#### Request datastorage to be mounted to dav

`POST` `/datastorage/{id}/davmount`

**Body**
```
{
  "time": 3600
}
```
`time` - mount time period in sec


#### Call off mounting to dav service for this datastorage
 `DELETE` `/datastorage/{id}/davmount`


### Workflow

 - User wants to mount some storage to dav and request it on datastorage page (new UI functionality)
 - When request is sent to server, metadata for this datastorage is updated with tag:
    ```
    dav-mount: <timestamp>
    ```
    **Note** If number of storages with `dav-mount` tag > then `system property` `storage.dav.mount.max.storages` user will get an error
    **Note** If storage already have `dav-mount` tag and new `timestamp` < then old one user will get an error


 - Dav service in the next scheduled synchronization will:
    - Load datastorage and check the tag, if dav-mount isn't persists - breaks
      - If timestamp is expired - remove tag from datastorage
      - Else mount this storage as before

**Comment by rodichenko** [comment]
@SilinPavel @mzueva @sidoruka let's introduce `storage.webdav.access.duration.seconds` preference for the *mount duration* in seconds:

- Group: **Data storage**
- Type: **STRING**
- Visible: **true**
- Default value: not set

**Comment by sidoruka** [comment]
@rodichenko, please add the following features as well:
* Add `ui.pipe.file.browser.request` preference of the following format:
```
{ 
    'start_request': 'Description of the Request Filesystem access feature',
    'done_request': 'Help tips - how to use the Filesystem access'
}
```
* If a `ui.pipe.file.browser.request` preference is specified:
   * If it contains `start_request` section - display a `help` icon with a tooltip next to `Request Filesystem Access` hyperlink:
![image](https://user-images.githubusercontent.com/20998291/140049066-fb9d4276-337b-4224-896e-3f2ecc9f5382.png)
  * If it  contains `done_request` section - display a `help` icon with a tooltip next to `File system access enabled`  label
  * If a preference is not set - no icons/tooltips are shown. If only one of the sections is missing - do not display an icon/tooltip for that state only.
  * Wordings for both tooltips shall be take from the corresponding preference section. It shall support MD formatiing

**Comment by maryvictol** [comment]
@rodichenko ,
**Bug:**
If attributes added for the data storage but *Request file system access* or *Configure notifications* isn't specified then click on *Remove all* button gives error `Could not delete non existing key.`. Attributes aren't removed.

**Comment by rodichenko** [comment]
> @rodichenko ,
> **Bug:**
> If attributes added for the data storage but _Request file system access_ or _Configure notifications_ isn't specified then click on _Remove all_ button gives error `Could not delete non existing key.`. Attributes aren't removed.

@maryvictol fixed with (dda521590696436965f997da28c2a8fd0bfa4867); "Remove all" button will ignore *special* attributes (like *Request file system access* or *Configure notifications*), even if they are set.

@sidoruka backported to `release/0.16`

**Comment by maryvictol** [comment]
@rodichenko 
Non-admin user doesn't see help icon with a tooltip next to *Request File system Access* hyperlink (or *File system access enabled* label)

**Comment by NShaforostov** [comment]
**Test cases** were created by #2313 and located [here](https://github.com/epam/cloud-pipeline/tree/develop/docs/testcases/UI/Data_storages/2233).

**Comment by maryvictol** [comment]
**e2e-tests** were implemented via #2336.

**Comment by maryvictol** [comment]
**Daily e2e-tests results** are located and updated [here](https://github.com/epam/cloud-pipeline/blob/e2e-reports/e2e/reports/AWS/test/classes/com.epam.pipeline.autotests.DavMountStorageLimitTest.html) (for `AWS`).
